### PR TITLE
Fix vulnerability

### PR DESCRIPTION
**Pull request by adkumar1**

### CODE REVIEW

File: .github/workflows/dapsreg-pipeline.yml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
           ghcr.io/catenax-ng/tx-daps-registration-service/dapsreg
         tags: |
           type=raw,value=latest,enable={{is_default_branch}}
-          type=semver,pattern={{version}},value=1.0.4
+          type=semver,pattern={{version}},value=1.0.5
```
Comment (carslen) [inline .github/workflows/dapsreg-pipeline.yml:56]: Stop hard coding image tags in your workflows!

File: Dockerfile [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 WORKDIR /drs
 
-RUN microdnf install dos2unix
+RUN microdnf install dos2unix && microdnf clean all
```
[the remark this answers is not in the document]
Comment (carslen) [reply Dockerfile:7]: Why executing dos2unix inside the container?! Please consider configuring your [local git configuration](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#_formatting_and_whitespace) appropiate.

**Review by carslen — CHANGES_REQUESTED** [review]
I won't accept hard coded image tagging in workflows any more. Please familiarize with releasing [images and tagging them](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-1).

Please also remove the dos2unix in the Dockerfile and fix line endings in github repo in a proper way, (please refer to comment).

File: charts/daps-reg-service/Chart.yaml [hunk 3]
```diff
@@ -32,10 +32,13 @@ description: A Helm chart for Kubernetes
 # pipeline. Library charts do not define any templates and therefore cannot be deployed.
 type: application
 
+sources:
+  - https://github.com/catenax-ng/tx-daps-registration-service
```
Comment (carslen) [inline charts/daps-reg-service/Chart.yaml:36]: This is not a valid source repository for Eclipse-Tractusx. Please use https://github.com/eclipse-tractusx/daps-registration-service instead.

Comment (carslen) [inline charts/daps-reg-service/Chart.yaml:36]: This is not a valid source repository for Eclipse-Tractusx. Please use https://github.com/eclipse-tractusx/daps-registration-service instead.

File: charts/daps-reg-service/values.yaml [hunk 4]
```diff
@@ -28,7 +28,7 @@ image:
   repository: "ghcr.io/catenax-ng/tx-daps-registration-service/dapsreg" 
   pullPolicy: Always
   # Overrides the image tag whose default is the chart appVersion.
-  tag: "1.0.4"
+  tag: "latest"
```
Comment (carslen) [inline charts/daps-reg-service/values.yaml:31]: Why using latest tag? It is not permitted (TRG 5.01, TRG 5.05).

**Review by carslen — CHANGES_REQUESTED** [review]

File: charts/daps-reg-service/values.yaml (hunk 4 above)
Comment (adkumar1) [reply charts/daps-reg-service/values.yaml]: Changed

File: charts/daps-reg-service/Chart.yaml (hunk 3 above)
Comment (adkumar1) [reply charts/daps-reg-service/Chart.yaml:36]: Changed

**Comment by adkumar1** [comment]
Can you push now

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (dvasunin) [reply Dockerfile]: dos2unix is used as the code for Docker image is built in the container, which may uses different EOL than host OS. The sources and build scripts are copied from host to the builder image, not checked out from git. If host is windows and we build Docker image from it we need to convert EOLs to the target format

[the remark this answers is not in the document]
Comment (carslen) [reply Dockerfile]: I am aware of the line ending problems on windows hosts. But what you describe applies to locally building the image on windows hosts, the image to use in the Helm chart is build by GH Workflow running ubuntu. 
From my point of view the `Dockerfile` shall not implement workarounds to grant correct file format on local builds. 

That's why I mentioned the local git configuration on your host. [There are ways](https://stackoverflow.com/questions/2517190/how-do-i-force-git-to-use-lf-instead-of-crlf-under-windows) (maybe this one is outdated due to age) to keep line ending in correct format on windows. Then you don't have to implement workarounds in `Dockerfile`.

**Review by carslen — CHANGES_REQUESTED** [review]
Please check review section in A1ODT-1207 and add more missing items to this PR.

File: CHANGELOG.md [hunk 5]
```diff
@@ -13,6 +13,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Created new helm charts for DAPS registration service
 - The latest version for daps registration service is 1.0.4
 
+## [1.0.4] - 2022-102-13
```
[the remark this answers is not in the document]
Comment (adkumar1) [reply CHANGELOG.md]: Taken care

**Comment by Siegfriedk** [comment]
Hi @adkumar1 ,

we had a short email with @wjost because we did not understand why this PR was merged after the discussions have not been resolved on this PR.

As he mentioned that you told him everything is adressed, i decided to revert this merge and as you to adress all discussions/conversatins on a PR.

**Comment by adkumar1** [comment]
Hi Sigi,

I have raised a new PR with the recommended correction. Please get it reviewed.

https://github.com/eclipse-tractusx/daps-registration-service/pull/14

--
Thanks & Regards,
Aditya Kumar



From: Sigi ***@***.***>
Date: Tuesday, 14 February 2023 at 4:28 PM
To: eclipse-tractusx/daps-registration-service ***@***.***>
Cc: Kumar, Aditya ***@***.***>, Mention ***@***.***>
Subject: Re: [eclipse-tractusx/daps-registration-service] Fix vulnerability (PR #12)

Hi @adkumar1<https://github.com/adkumar1> ,

we had a short email with @wjost<https://github.com/wjost> because we did not understand why this PR was merged after the discussions have not been resolved on this PR.

As he mentioned that you told him everything is adressed, i decided to revert this merge and as you to adress all discussions/conversatins on a PR.

**Comment by carslen** [comment]
I reverted the commit of the triggered [workflow run](https://github.com/eclipse-tractusx/daps-registration-service/actions/runs/4171446134) and deleted the released HELM Chart version 1.0.4 (never decrease the chart version) as well as the corresponding GH tag.

**Comment by adkumar1** [comment]
Please consider it for the first time the decreased version.
As the application version is 1.0.4, so our team wants to have everything in the same version
Going forward will take care of this thing

**Comment by carslen** [comment]
There is already a released helm chart with appVersion 1.0.4:

```
$ helm repo list
NAME        	URL
tractusx-dev	https://eclipse-tractusx.github.io/charts/dev
$ helm search repo daps-reg --versions
NAME                         	CHART VERSION	APP VERSION	DESCRIPTION
tractusx-dev/daps-reg-service	1.0.5        	1.0.4      	A Helm chart for Kubernetes
```

If you want to keep Chart version and AppVersion in sync, implement it with the next appVersion. Your Helm Chart improved, got changed. This is the reason why the Chart version should be increased, instead of decreased. No one would ever use your improved Helm Chart with the decreased helm chart version, as everyone would expect the improvements in the higher versioned helm chart.

**Comment by adkumar1** [comment]
Can't it be possible to have the application version same as the Releases ? Because people refer the version no. here in the releases and get confused with the application version .
[Releases] -> daps-reg-service-1.0.4

**Comment by carslen** [comment]
The Helm chart doesn't represent the product, but an installer for a product. If you look at [Bitnamis Postgresql](https://github.com/bitnami/charts/blob/main/bitnami/postgresql/Chart.yaml) or [ingress-nginx](https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/Chart.yaml) Helm chart, you can see that Chart Version and AppVersion do not match. You always have to check the Helm chart to get an idea of the application version it will install.

**Pull request merged**